sql/core/src/test/scala/org/apache/spark/sql/SparkSessionJobTaggingAndCancellationSuite.scala

```diff
@@ -330,4 +330,109 @@ class SparkSessionJobTaggingAndCancellationSuite
       threadPool.shutdownNow()
     }
   }
+
+  test("Interrupt jobs by tag") {
```

Comment:
Hi, @xupefei . I add two test cases here to ensure consistency with testing on the PySpark side (which are defined `python/pyspark/sql/tests/test_job_cancellation.py` above) but seems like they are not working as I expected.

Could you take a look at these new tests to confirm if I missed something or should we actually made a fix for core side first?? Thanks!

Also cc @HyukjinKwon

Reply:
After [this PR](https://github.com/apache/spark/pull/48906) tags are isolated per session: tags added my the main thread can be seen from child threads.

In this test case, `session.interruptTag("tag_A")` (which runs in the main thread) won't see `session.addTag(tag)` (runs in a child* thread).

*) Actually when working on the aforementioned PR, I found that threads from a Pool are not considered a "child" thread and won't see tags. I have to manually create threads instead: https://github.com/apache/spark/pull/48906/files#diff-647cd00e281c3cccb76a42646a692abaef6738c8157daf6642bc8fa16e233394R168

Reply:
Hi @xupefei could you check my last comment above when you have some time?

I wonder if we should fix the Scala side logic or should we just fix the test? (btw this test scenario is working properly from Spark Connect Python Client so I think anyway we should match the behavior)

also cc @HyukjinKwon fyi